Framework/DataHandling/src/LoadEventNexus.cpp

```diff
@@ -447,33 +460,23 @@ firstLastPulseTimes(::NeXus::File &file, Kernel::Logger &logger) {
           units);
     std::vector<uint64_t> nanoseconds;
     file.readData("event_time_zero", nanoseconds);
-    if (nanoseconds.empty())
-      throw std::runtime_error(
-          "No event time zeros. Cannot establish run start or end");
     auto absoluteFirst = DateAndTime(int64_t(0), int64_t(nanoseconds.front())) +
                          offset.totalNanoseconds();
-    auto absoluteLast = DateAndTime(int64_t(0), int64_t(nanoseconds.back())) +
-                        offset.totalNanoseconds();
-    return std::make_pair(absoluteFirst, absoluteLast);
+    return absoluteFirst;
   } else if (heldTimeZeroType == ::NeXus::FLOAT64) {
     if (units != "second")
       logger.warning("event_time_zero is double_t, but units not in seconds. "
                      "Found to be: " +
                      units);
     std::vector<double> seconds;
     file.readData("event_time_zero", seconds);
-    if (seconds.empty())
-      throw std::runtime_error(
-          "No event time zeros. Cannot establish run start or end");
     auto absoluteFirst =
         DateAndTime(seconds.front(), double(0)) + offset.totalNanoseconds();
-    auto absoluteLast =
-        DateAndTime(seconds.back(), double(0)) + offset.totalNanoseconds();
-    return std::make_pair(absoluteFirst, absoluteLast);
-  }
-
-  else {
-    throw std::runtime_error("Unrecognised type for event_time_zero");
+    return absoluteFirst;
+  } else {
+    std::string message = "Unrecognised type for event_time_zero";
+    logger.warning(message);
+    throw std::runtime_error(message);
```

Comment:
Warning then throw.
Also is it possible to report what the actual held type is. That would help future maintainers.